src/low-level/imap/mailimap_types.h

```diff
@@ -3341,6 +3341,8 @@ enum {
 
 typedef void mailimap_msg_att_handler(struct mailimap_msg_att * msg_att, void * context);
 
+typedef bool mailimap_msg_body_handler(const char *str, size_t len, void * context);
```

Comment:
Could you add:
int msg_att_type
(possible values: MAILIMAP_MSG_ATT_BODY_SECTION, MAILIMAP_MSG_ATT_RFC822_TEXT, MAILIMAP_MSG_ATT_RFC822_HEADER, MAILIMAP_MSG_ATT_RFC822)
and struct mailimap_msg_att_body_section * section
to the parameters?

I think you would need to add those to mailimap_parser_context as well.